core-rust/state-manager/src/state_manager.rs

```diff
@@ -376,14 +376,16 @@ where
                 .wrap(executable, ConfigType::Genesis)
                 .warn_after(TRANSACTION_RUNTIME_WARN_THRESHOLD, logged_description),
         );
-
         let commit = processed.expect_commit(logged_description);
         commit.check_success(logged_description);
+        let validator_set = commit
+            .next_epoch()
+            .map(|next_epoch| next_epoch.validator_set);
+        let ledger_hashes = commit.hash_structures_diff.ledger_hashes;
```

Comment:
(note from author)

In this PR, you will see a couple of "unnecessary" re-orders like this one.
I did them to clearly see where the mutex on `execution_cache` can be released (coming soon).